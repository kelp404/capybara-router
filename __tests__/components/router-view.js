const history = require('history');
const React = require('react');
const renderer = require('react-test-renderer');
const {Router, RouterView} = require('../../');

let router;
beforeEach(() => {
  router = new Router({
    history: history.createMemoryHistory({initialEntries: ['/']}),
    routes: [
      {name: 'home', uri: '/'}
    ]
  });
  router.start();
});

test('RouterView component render.', () => {
  const component = renderer.create(<RouterView>Loading</RouterView>);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('RouterView component render with a child component.', () => {
  class Child extends React.Component {
    render() {
      // eslint-disable-next-line react/prop-types
      return <div className={this.props.className}>child</div>;
    }
  }

  let routerView;
  router.registerRouterView = jest.fn(view => {
    routerView = view;
  });
  const component = renderer.create(<RouterView>Loading</RouterView>);
  routerView.dispatch({
    route: {component: Child},
    props: {className: 'head'}
  });
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('RouterView component will call router.registerRouterView() on the mount event.', () => {
  router.registerRouterView = jest.fn(() => {});
  renderer.create(<RouterView>Loading</RouterView>);
  expect(router.registerRouterView).toBeCalled();
});
