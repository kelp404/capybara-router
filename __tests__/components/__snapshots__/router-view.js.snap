// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RouterView component render with a child component. 1`] = `
<div
  class="head"
>
  child
</div>
`;

exports[`RouterView component render. 1`] = `
<div>
  Loading
</div>
`;
